File: php/testcode/ssrf_002.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_ssrf_curl_hardcoded
function ssrf_curl_hardcoded(BenchmarkRequest $req): BenchmarkResponse {
    $ch = curl_init();
    curl_setopt($ch, CURLOPT_URL, "https://api.internal.com/status"); // vuln-code-snippet safe-line php_ssrf_curl_hardcoded
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    $response = curl_exec($ch);
    curl_close($ch);
    return BenchmarkResponse::ok($response);
}
// vuln-code-snippet end php_ssrf_curl_hardcoded

File: ruby/testcode/ssrf_002.rb
require_relative 'shared'
require 'net/http'
require 'uri'

ALLOWED_DOMAINS = %w[api.example.com cdn.example.com feeds.example.com].freeze

# vuln-code-snippet start ruby_ssrf_allowlist_domain
def proxy_request(req)
  url = req.param('url')
  uri = URI.parse(url)
  unless ALLOWED_DOMAINS.include?(uri.host) # vuln-code-snippet safe-line ruby_ssrf_allowlist_domain
    return BenchmarkResponse.bad_request("domain not allowed")
  end
  response = Net::HTTP.get(uri)
  BenchmarkResponse.ok(response)
end
# vuln-code-snippet end ruby_ssrf_allowlist_domain
